charms/k8s/lib/charms/k8s/v0/k8sd_api_manager.py

```diff
@@ -302,7 +403,7 @@ def create_join_token(self, name: str):
         """
         endpoint = "/1.0/k8sd/tokens"
```

Comment:
The tokens endpoint was removed in https://github.com/canonical/k8s-snap/pull/73
The additional logic around the token generation moved into the microcluster hooks which made this endpoint obsolete. 

Instead, directly query the microcluster endpoint `cluster/1.0/tokens` to issue a new token.
See: 
https://github.com/canonical/microcluster/blob/1525f8ea8d7a2196d7825bc3ca5fc29368a0fd1b/internal/rest/client/tokens.go#L18
and the type definition here:
https://github.com/canonical/microcluster/blob/1525f8ea8d7a2196d7825bc3ca5fc29368a0fd1b/internal/rest/types/tokens.go#L11